src/index.js

```diff
@@ -15,11 +25,94 @@ console.log(
 // 자바스크립트 코드에서 이미지 리소스 로드 테스트
 // index.html 파일의 html 구조를 수정하셔도 됩니다.
 addEventListener("load", () => {
-  const app = document.querySelector("#app");
-  const buttonImage = document.createElement("img");
-  buttonImage.src = image;
+  let categoryFilterValue = INIT_CATEGORY_FILTER_VALUE;
+  let sortingFilterValue = SORTING_FILTER_VALUE;
+
+  const app = document.querySelector("main");
+  app.prepend(createHeaderComponent());
+
+  const section = document.querySelector(".restaurant-filter-container");
+  section.appendChild(
+    createSelectComponent({
+      attName: "category",
+      className: "restaurant-filter",
+      id: "category-filter",
+      options: RESTAURANT_CATEGORY_LIST,
+    })
+  );
+
+  section.appendChild(
+    createSelectComponent({
+      attName: "sorting",
+      className: "restaurant-filter",
+      id: "sorting-filter",
+      options: SORT_OPTION_LIST,
+    })
+  );
+
+  const restaurantList = document.querySelector(".restaurant-list");
+
+  renderRestaurantList({
+    restaurantList,
+    sortingFilterValue,
+    categoryFilterValue,
+  });
 
-  if (app) {
-    app.appendChild(buttonImage);
-  }
+  document.addEventListener("change", (e) => {
+    if (e.target.id === "category-filter") {
+      categoryFilterValue = e.target.value;
+      renderRestaurantList({
+        restaurantList,
+        sortingFilterValue,
+        categoryFilterValue,
+      });
+    }
+    if (e.target.id === "sorting-filter") {
+      sortingFilterValue = e.target.value;
+      renderRestaurantList({
+        restaurantList,
+        sortingFilterValue,
+        categoryFilterValue,
+      });
+    }
+  });
 });
+
+function renderRestaurantList({
+  restaurantList,
+  sortingFilterValue,
+  categoryFilterValue,
+}) {
+  restaurantList.innerHTML = "";
+  const restaurants = filteredRestaurants(categoryFilterValue);
+
+  sortedRestaurants(restaurants, sortingFilterValue);
+
+  restaurants.forEach((restaurant) => {
+    restaurantList.appendChild(createRestaurantListComponent(restaurant));
+  });
+}
+
+function filteredRestaurants(categoryFilterValue) {
+  return RESTAURANT_LIST.filter((item) =>
+    categoryFilterValue === "전체"
+      ? true
+      : item.category === categoryFilterValue
```

Comment:
이렇게 하거나,
```suggestion
    categoryFilterValue === "전체" || categoryFilterValue === item.category
```
아래와 같이 한다면
```suggestion
    ["전체", item.category].includes(categoryFilterValue)
```
동일하게 작성하되 좀 더 직관적으로 갈 수 있을 것 같네요!